TestLib/Exception/GREYDefaultFailureHandler.m

```diff
@@ -81,23 +94,46 @@ - (void)handleException:(GREYFrameworkException *)exception details:(NSString *)
       [GREYFailureScreenshotSaver saveFailureScreenshotsInDictionary:appScreenshots
                                                          toDirectory:screenshotDir];
   NSAssert(screenshotPaths, @"Screenshots must be present");
+  return screenshotPaths;
+}
 
-  NSString *appUIHierarchy = [exception.userInfo valueForKey:kErrorDetailAppUIHierarchyKey];
-  // For calls from GREYAsserts in the test side, the hierarchy must be populated here.
-  if (!appUIHierarchy) {
-    appUIHierarchy = [GREYElementHierarchy hierarchyString];
+- (NSString *)consoleLogForException:(GREYFrameworkException *)exception
+                             details:(NSString *)details
+                     screenshotPaths:(GREYFailureScreenshots *)screenshotPaths
+                      appUIhierarchy:(NSString *)appUIHierarchy
```

Comment:
You don't need to pass the appUIHierarchy and screenshots here as they are computed only in this method. You can move the calls to 

```
GREYFailureScreenshots *screenshotPaths = [self screenshotPathsForException:exception];
NSString *appUIHierarchy = [self appUIHierarchyForException:exception];
```

Inside this function where there are being used. So, `[self appUIHierarchyForException:exception];` goes where the formatFailure... method is called.

Reply:
Resolved